Plugin/RenderingCacheContextPlugin.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Plugin;
+
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Magento\Framework\App\Http\Context as HttpContext;
+use Magento\Framework\App\Request\Http;
+use Magento\Store\Model\StoreManagerInterface;
+
+class RenderingCacheContextPlugin
```

Comment:
Nice job, on the plugin! 👏 
I suppose I know what and how it's doing, but could you elaborate more on its' functionality and give examples how it kicks in when a user or robot sees the page? Thank you!